### PR TITLE
Replace int with bool in comparisons of link layer addresses

### PR DESCRIPTION
**Pull request by kkrentz**
Small improvement to use the bool datatype in comparisons of link layer addresses.

### CODE REVIEW

File: os/net/packetbuf.h [hunk 1]
```diff
@@ -268,7 +268,7 @@ const linkaddr_t *packetbuf_addr(uint8_t type);
  * \brief      Checks whether the current packet is a broadcast.
  * \retval 0   iff current packet is not a broadcast
```
Comment (pjonsson) [inline os/net/packetbuf.h:269]: Documentation should also be updated if the return types are changed.

Comment (kkrentz) [reply os/net/packetbuf.h]: good proposal

File: os/net/packetbuf.h [hunk 2]
```diff
@@ -265,10 +265,10 @@ int               packetbuf_set_addr(uint8_t type, const linkaddr_t *addr);
 const linkaddr_t *packetbuf_addr(uint8_t type);
 
 /**
- * \brief      Checks whether the current packet is a broadcast.
- * \retval 0   iff current packet is not a broadcast
+ * \brief       Checks whether the current packet is a broadcast.
+ * \retval true iff the current packet is not a broadcast
```
Comment (nvt) [inline os/net/packetbuf.h:269]: Now the comment says that the function returns <code>true</code> "iff the current packet is _not_ a broadcast".

Comment (kkrentz) [reply os/net/packetbuf.h]: Fixed comment

**Review by nfi — APPROVED** [review]

**Pull request merged**